server/src/testFixtures/java/io/crate/testing/TestingHelpers.java

```diff
@@ -436,4 +438,20 @@ public static void assertCrateVersion(Object object, Version versionCreated, Ver
                     is(Version.Property.UPGRADED.toString()),
                     versionUpgraded == null ? nullValue() : is(versionUpgraded.externalNumber()))));
     }
+
+    public static <T> List<T> getRandomsOfType(int minLength, int maxLength, DataType<T> dataType) {
+        var values = new ArrayList<T>();
+
+        boolean addNull = RandomizedTest.randomBoolean();
+        int length  = RandomizedTest.randomIntBetween(minLength, maxLength);
+        if (addNull) {
+            length = length - 1;
+            values.add(null);
+        }
+        var generator = DataTypeTesting.getDataGenerator(dataType);
+        for (int i = 0; i < length; i++) {
+            values.add(dataType.sanitizeValue(generator.get()));
+        }
```

Comment:
What do you think of implementing it like this:


```suggestion
        var generator = DataTypeTesting.getDataGenerator(dataType);
        for (int i = 0; i < length; i++) {
            // 1/length chance
            if (RandomizedTest.randomInt(length) == 0) {
              values.add(null);
            } else {
              values.add(dataType.sanitizeValue(generator.get()));
            }
        }
```

This should still result in approximately 1 null value per array, but randomizes the position of it. So that it is not always at the start or end, but can be anywhere.

Reply:
Oh, sorry, I saw that comment after adding ready-to-merge label. I can do this in array_sum PR.